internal/driver/glfw/canvas_test.go

```diff
@@ -348,24 +348,31 @@ func TestGlCanvas_MinSizeShrinkTriggersLayout(t *testing.T) {
 	w.SetContent(content)
 
 	oldCanvasSize := fyne.NewSize(200+3*theme.Padding(), 100+3*theme.Padding())
-	assert.Equal(t, oldCanvasSize, c.Size())
+	runOnMain(func() {
+		assert.Equal(t, oldCanvasSize, c.Size())
```

Comment:
This doesn't make sense as `c.Size()` is on safeCanvas and already on main

Reply:
Ah, it did not deadlock because you have a check for gorutine ID in there. Will fix!

Reply:
Fixed it up just now